client/src/components/ReviewSection.jsx

```diff
@@ -1,43 +1,119 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import RatingsBreakdown from './RatingBreakdown.jsx';
 import ReviewList from './ReviewList.jsx';
 import AddReviewModal from './AddReviewModal.jsx';
 
-class ReviewSection extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showAddReviewModal: false,
-      reviews: []
-    };
-  }
-
-  showAddReviewModal() {
-    this.setState({showAddReviewModal: true});
-  }
-
-  closeAddReviewModal() {
-    this.setState({showAddReviewModal: false});
-  }
-
-  fetchReviews() {
-
-  }
-
-  render() {
-
-    // characteristics={Object.keys(this.reviewData.meta.characteristics)}
-    return (
-      <div data-testid="reviewSection" className="reviewSection">
-        {this.state.showAddReviewModal ? <AddReviewModal itemName={this.props.itemInfo.name} handleClose={this.closeAddReviewModal.bind(this)}></AddReviewModal> : null}
-        <span className="widgetHeader">Ratings &#38; Reviews</span>
-        <div id="mainReviewSection">
-          <RatingsBreakdown metaData={this.props.reviewData ? this.props.reviewData.meta : false}/>
-          <ReviewList reviews={this.props.reviewData ? this.props.reviewData.allReviews : false} showAddReviewModal={this.showAddReviewModal.bind(this)}/>
-        </div>
+const ReviewSection = props => {
+  const [reviewModalDisplay, setReviewModalDisplay] = useState(false);
+  const [reviews, setReviews] = useState([]);
+  const [reviewsMetaData, setReviewsMetaData] = useState({});
+  const [sortOption, setSortOption] = useState('relevant');
+  const [filters, setFilters] = useState({});
+
+  const [filteredReviews, setFilteredReviews] = useState([]);
+
+
+  const showAddReviewModal = () => {
+    setReviewModalDisplay(true);
+  };
+
+  const closeAddReviewModal = () => {
+    setReviewModalDisplay(false);
+  };
+
+  const handleSortChange = (sortOption) => {
+    setSortOption(sortOption);
+  };
+
+  const handleFilterChange = (target) => {
+    let rating = target.innerHTML[0];
+    let updatedFilters = filters;
+    updatedFilters[rating] ? delete updatedFilters[rating] : updatedFilters[rating] = true;
```

Comment:
I didn't realize you could use turneries in a way that didn't resolve to some sort of value, but that's handy to know!